src/utilities/include/par.h

```diff
@@ -592,15 +803,41 @@ OutputIter copy_if(ExecutionPolicy policy, InputIter first, InputIter last,
   return std::copy_if(first, last, d_first, pred);
 }
 
+// Copy values in the input range `[first, last)` to the output range
+// starting from `d_first` that satisfies the predicate `pred`, i.e. `pred(x) ==
+// true`, and returns `d_first + n` where `n` is the number of times the
+// predicate is evaluated to true.
+//
+// This function is stable, meaning that the relative order of elements in the
+// output range remains unchanged.
+//
+// The input range `[first, last)` and
+// the output range `[d_first, d_first + last - first)`
+// must be equal or non-overlapping.
```

Comment:
Does this really work in-place (equal)? And if so, does `remove_if` really need that `tmp` vector?

Reply:
good cach, I copied the wrong lines.